cobalt/browser/global_features.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_BROWSER_GLOBAL_FEATURES_H_
+#define COBALT_BROWSER_GLOBAL_FEATURES_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/no_destructor.h"
+#include "base/sequence_checker.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "content/shell/browser/shell_paths.h"
+
+class PrefService;
+
+namespace metrics {
+class MetricsService;
+}  // namespace metrics
+
+namespace metrics_services_manager {
+class MetricsServicesManager;
+}  // namespace metrics_services_manager
+
+namespace cobalt {
+class CobaltMetricsServicesManagerClient;
+
+extern const char kExperimentConfigFeature[];
+extern const char kExperimentConfigFeatureParams[];
+extern const char kExperimentConfigExpIds[];
+
+// This class owns features that are globally scoped. It follows the structure
+// of BrowserProcess class in Chrome.
```

Comment:
I like this comment. Always try to specify lifetime and threading too, e.g. `This class is a singleton and can only be accessed on a single SequenceTaskRunner`.